### PR TITLE
Fix WP_MOK hardcoded to 8 bytes, use storage_size for portability

### LINKED ISSUE #1207: WP_MOK hardcoded to 8 bytes, breaks single-precision builds
## Bug

Multiple files compute \`WP_MOK = int(8._wp, MPI_OFFSET_KIND)\`, hardcoding 8 bytes as the working precision real size. When MFC is built in single precision (\`wp = real32\`), reals are 4 bytes, but MPI I/O operations use an 8-byte offset per element.

**Files:** \`src/post_process/m_data_input.f90\`, \`src/pre_process/m_data_output.fpp\`, \`src/pre_process/m_start_up.fpp\`, \`src/simulation/m_data_output.fpp\`, \`src/simulation/m_start_up.fpp\` (10 occurrences total)

**Impact:** MPI parallel I/O reads/writes the wrong number of bytes per element in single-precision builds, corrupting all file I/O.

**Severity:** High — all parallel I/O is broken in single-precision builds.

### PR DESCRIPTION
**Pull request by sbryngelson**
## Summary

**Severity:** HIGH — MPI file I/O reads from wrong offsets in single-precision builds.

`WP_MOK` (the MPI offset for the working-precision real size) is hardcoded to `int(8._wp, MPI_OFFSET_KIND)`, assuming 8-byte (double-precision) reals. In single-precision builds where `wp` corresponds to 4 bytes, this causes all MPI file read offsets to be wrong by a factor of 2.

### Before
```fortran
WP_MOK = int(8._wp, MPI_OFFSET_KIND)    ! assumes double precision
```

### After
```fortran
WP_MOK = int(storage_size(0._stp)/8, MPI_OFFSET_KIND)    ! actual byte size
```

All 10 occurrences across 5 files are fixed. One site in `src/simulation/m_start_up.fpp` had `int(4._wp, ...)` instead of `int(8._wp, ...)` — wrong in both single and double precision.

Additionally, 3 sites in pre_process used `storage_size(0._wp)` instead of `storage_size(0._stp)` — since field arrays use storage precision (`stp`), the I/O byte stride must match `stp`, not `wp`.

### Why this went undetected
The codebase is predominantly used in double precision where `wp` = `stp` = 8 bytes, so the hardcoded value happens to be correct.

## Also fixes: `dq_prim_dy_vf` → `dq_prim_dz_vf` bug in 2D cylindrical viscous boundary

In `src/simulation/m_rhs.fpp`, the call to `s_compute_viscous_stress_cylindrical_boundary` in the `p == 0` (2D) branch was passing `dq_prim_dy_vf` as the z-gradient argument (`grad_z_vf`) instead of `dq_prim_dz_vf`. The `p > 0` (3D) branch was correct. This was a duplicate-line copy-paste error that would cause incorrect viscous stress computation at cylindrical boundaries in 2D simulations.

## Files changed
- `src/post_process/m_data_input.f90` — 3 WP_MOK sites
- `src/pre_process/m_data_output.fpp` — 2 WP_MOK sites
- `src/pre_process/m_start_up.fpp` — 1 WP_MOK site
- `src/simulation/m_data_output.fpp` — 3 WP_MOK sites
- `src/simulation/m_start_up.fpp` — 1 WP_MOK site
- `src/simulation/m_rhs.fpp` — 1 `dq_prim_dy_vf` → `dq_prim_dz_vf` fix

## Test plan
- [ ] Build in both single and double precision
- [ ] Verify MPI parallel I/O reads correct data in both modes
- [ ] Verify 2D cylindrical viscous boundary cases produce correct results

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Fixes #1207

### CODE REVIEW

File: src/post_process/m_data_input.f90 [hunk 1]
```diff
@@ -177,7 +177,7 @@ impure subroutine s_read_ib_data_files(file_loc_base, t_step)
                 n_MOK = int(n_glb + 1, MPI_OFFSET_KIND)
                 p_MOK = int(p_glb + 1, MPI_OFFSET_KIND)
                 MOK = int(1._wp, MPI_OFFSET_KIND)
-                WP_MOK = int(8._wp, MPI_OFFSET_KIND)
+                WP_MOK = int(storage_size(0)/8, MPI_OFFSET_KIND)
```
Comment (wilfonba) [inline src/post_process/m_data_input.f90]: missing `._stp`

File: src/pre_process/m_data_output.fpp [hunk 2]
```diff
@@ -549,7 +549,7 @@ contains
             m_MOK = int(m_glb_save, MPI_OFFSET_KIND)
             n_MOK = int(n_glb_save, MPI_OFFSET_KIND)
             p_MOK = int(p_glb_save, MPI_OFFSET_KIND)
-            WP_MOK = int(8._wp, MPI_OFFSET_KIND)
+            WP_MOK = int(storage_size(0._wp)/8, MPI_OFFSET_KIND)
```
Comment (wilfonba) [inline src/pre_process/m_data_output.fpp]: `._stp`?

File: src/pre_process/m_data_output.fpp [hunk 3]
```diff
@@ -615,7 +615,7 @@ contains
             m_MOK = int(m_glb + 1, MPI_OFFSET_KIND)
             n_MOK = int(n_glb + 1, MPI_OFFSET_KIND)
             p_MOK = int(p_glb + 1, MPI_OFFSET_KIND)
-            WP_MOK = int(8._wp, MPI_OFFSET_KIND)
+            WP_MOK = int(storage_size(0._wp)/8, MPI_OFFSET_KIND)
```
Comment (wilfonba) [inline src/pre_process/m_data_output.fpp]: `._stp`?

File: src/pre_process/m_start_up.fpp [hunk 4]
```diff
@@ -660,7 +660,7 @@ contains
             m_MOK = int(m_glb + 1, MPI_OFFSET_KIND)
             n_MOK = int(n_glb + 1, MPI_OFFSET_KIND)
             p_MOK = int(p_glb + 1, MPI_OFFSET_KIND)
-            WP_MOK = int(8._wp, MPI_OFFSET_KIND)
+            WP_MOK = int(storage_size(0._wp)/8, MPI_OFFSET_KIND)
```
Comment (wilfonba) [inline src/pre_process/m_start_up.fpp]: `._stp`?

File: src/simulation/m_data_output.fpp [hunk 5]
```diff
@@ -1087,7 +1087,7 @@ contains
         m_MOK = int(m_glb + 1, MPI_OFFSET_KIND)
         n_MOK = int(n_glb + 1, MPI_OFFSET_KIND)
         p_MOK = int(p_glb + 1, MPI_OFFSET_KIND)
-        WP_MOK = int(8._wp, MPI_OFFSET_KIND)
+        WP_MOK = int(storage_size(0)/8, MPI_OFFSET_KIND)
```
Comment (wilfonba) [inline src/simulation/m_data_output.fpp]: missing `._stp`

**Comment by anandrdbz** [comment]
This is correct, change should be added
